src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/StateFlags.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Text.RegularExpressions.Symbolic
+{
+    /// <summary>
+    /// These flags provide context-independent information available for every state. They provide a fast way to evaluate
+    /// conditions in the inner matching loops of <see cref="SymbolicRegexMatcher{TSet}"/>. The matcher caches one of these
+    /// for every state, for which they are created by <see cref="MatchingState{TSet}.BuildStateFlags(ISolver{TSet}, bool)"/>.
+    /// In DFA mode the cached flags are used directly, while in NFA mode the <see cref="SymbolicRegexMatcher{TSet}.NfaStateHandler"/>
+    /// handles aggregating the flags in the state set.
+    /// </summary>
+    [Flags]
+    internal enum StateFlags : byte
+    {
+        IsInitialFlag = 1,
+        IsDeadendFlag = 2,
+        IsNullableFlag = 4,
+        CanBeNullableFlag = 8,
+        SimulatesBacktrackingFlag = 16,
+    }
+
+    /// <summary>
+    /// These extension methods for <see cref="StateFlags"/> make checking for the presence of flags more concise.
+    /// </summary>
+    internal static class StateFlagsExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I assume this is going to get inlined anyway, but I have seen us add AggressiveInlining even when we know it will get inlined so it is probably fine to keep it.

Reply:
I removed them. I'll check performance isn't affected before merging!